### PR TITLE
Port elements-sk to TypeScript

### PR DESCRIPTION
**Pull request by LeandroLovisolo**
In this PR:

- Code has been moved from `//elements-sk` to `//src`.
- `//elements-sk` is now generated with `npm run build`.
- `npm run build` compiles TypeScript sources with `tsc` (reads `//src/**/*.ts`, outputs `//elements-sk/**/*.js` along with maps, type definitions, etc.)
- `npm run build` copies Non-TypeScript assets such as stylesheets from `//src` to `//elements-sk` as a post-compilation step.
- In order to publish to npm, it's now necessary to first run `npm run build`.
- Karma executes TypeScript tests directly via the [karma-typescript](https://www.npmjs.com/package/karma-typescript) plugin.
- Webpack is now only used to generate the demo pages.

### CODE REVIEW

File: generate_icons.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ import (
 )
 
 const SVG_SRC = "./node_modules/material-design-icons/%s/svg/production/"
-const SVG_DEST = "./elements-sk/icon/%s-icon-sk.js"
+const SVG_DEST = "./src/icon/%s-icon-sk.ts"
```
Comment (jcgregorio) [inline generate_icons.go]: Might want to move these const names to lowerCamelCase.

File: src/checkbox-sk/checkbox-sk.ts [hunk 2]
```diff
@@ -51,16 +51,20 @@ import { define } from '../define';
 import { upgradeProperty } from '../upgradeProperty';
```
Comment (jcgregorio) [inline src/checkbox-sk/checkbox-sk.ts:51]: Per discussion on the other PR, the type annotations can be removed from all JSDoc.

File: src/checkbox-sk/checkbox-sk_test.ts [hunk 3]
```diff
@@ -22,17 +24,17 @@ afterEach(() => {
   container.innerHTML = '';
 });
 
-function _checkBoxSetup() {
+function _checkBoxSetup(): Promise<CheckOrRadio> {
   return window.customElements.whenDefined('checkbox-sk').then(() => {
```
Comment (jcgregorio) [inline src/checkbox-sk/checkbox-sk_test.ts]: These could also be changed to async/await, not sure if that would be any more readable.

File: generate_icons.go (hunk 1 above)
Comment (LeandroLovisolo) [reply generate_icons.go]: Done.

File: src/checkbox-sk/checkbox-sk.ts (hunk 2 above)
Comment (LeandroLovisolo) [reply src/checkbox-sk/checkbox-sk.ts:51]: Done.

File: src/checkbox-sk/checkbox-sk_test.ts (hunk 3 above)
Comment (LeandroLovisolo) [reply src/checkbox-sk/checkbox-sk_test.ts]: 100% agree that async/await is more readable.

**Review by jcgregorio — APPROVED** [review]
LGTM

**Pull request merged**